src/mdarray/mdarray.jl

```diff
@@ -0,0 +1,372 @@
+abstract type AbstractExtendedDataType end
+# needs to have a `ptr::GDALExtendedDataTypeH` attribute
+
+abstract type AbstractEDTComponent end
+# needs to have a `ptr::GDALEDTComponentH` attribute
+
+abstract type AbstractGroup end
+# needs to have a `ptr::GDAL.GDALGroupH` attribute
+
+abstract type AbstractMDArray end
```

Comment:
We probably want this to be <: AbstractDiskArray so chunked reads work automatically.